### PR TITLE
feat: adds no-extra-spacing-text

### PR DESCRIPTION
**Pull request by RobertAKARobin**
This adds a rule to prevent unnecessary whitespace in text.

### CODE REVIEW

File: packages/eslint-plugin/lib/rules/no-extra-spacing-text.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        checkSiblings(node.body);
+      },
+    };
+
+    /**
+     * @param {TextNode | CommentContentNode} node
+     */
+    function stripConsecutiveSpaces(node) {
+      const text = node.value;
+      const matcher = /(^|[^\n \t])([ \t]{2,})($|[^\n \t])/g;
```
Comment (yeonjuan) [inline packages/eslint-plugin/lib/rules/no-extra-spacing-text.js]: This regex doesn't seem to match trailing spaces. Is it possible to handle trailing whitespace as well?
```js
// spaces exists after the span
    {
      code: `
<div>
   <span></span>  
</div>
      `,
      output: `
<div>
   <span></span>
</div>
      `,
      errors: errorsAt([3, 16, 2]),
    },
  ],
```
스크린샷 2024-09-04 오전 1 11 02

Comment (RobertAKARobin) [reply packages/eslint-plugin/lib/rules/no-extra-spacing-text.js]: @yeonjuan  Added!

**Review by yeonjuan — APPROVED** [review]
LGTM!! Thank you for doing this work.

**Comment by TheElegantCoding** [comment]
Hello i dont see this rule in the website, the doc is updated ?

**Pull request merged**